apps/frontend/src/libs/helpers/get-permitted-navigation-items/get-permitted-navigation-items.helper.ts

```diff
@@ -1,16 +1,23 @@
+import { EMPTY_LENGTH } from "~/libs/constants/constants.js";
 import {
 	type NavigationItem,
 	type PermissionGetAllItemResponseDto,
+	type ProjectPermissionsGetAllItemResponseDto,
 } from "~/libs/types/types.js";
 
 import { checkHasPermission } from "../helpers.js";
 
 const getPermittedNavigationItems = (
 	items: NavigationItem[],
-	userPermissions: PermissionGetAllItemResponseDto[],
+	userPermissions: PermissionGetAllItemResponseDto[] = [],
+	projectPermissions: ProjectPermissionsGetAllItemResponseDto[] = [],
 ): NavigationItem[] => {
-	return items.filter(({ pagePermissions = [] }) =>
-		checkHasPermission(pagePermissions, userPermissions),
+	return items.filter(
+		({ pagePermissions = [], pageProjectPermissions = [] }) =>
+			(pagePermissions.length !== EMPTY_LENGTH &&
+				checkHasPermission(pagePermissions, userPermissions)) ||
+			(pageProjectPermissions.length !== EMPTY_LENGTH &&
+				checkHasPermission(pageProjectPermissions, projectPermissions)),
```

Comment:
use curly braces and split this checks into constants inside it, so that it looks more readable